### ISSUE TITLE
Gitolite::Config::GroupDependencyError

### ISSUE THREAD

**Issue by mmcc** [wingrunr21/gitolite#17]
Working on a script that uses Gitolite I ran into a strange GroupDependencyError problem. It would spit out the error and wipe my config file completely. I figured it was a problem with my script, but then when I try to simply add a repo through irb I get the same issue. Here's the sequence I'm running:
1. `require 'gitolite'`
2. `ga_repo = Gitolite::GitoliteAdmin.new("./gitolite-admin")`
3. `conf = ga_repo.config`
4. `repo = Gitolite::Config::Repo.new("test")`
5. `repo.add_permission("RW+", "", "@plexus")`
6. `conf.add_repo(repo)`
7. `ga_repo.save`

All of these steps are successful up until step 7, at which point I get this error.

```
Gitolite::Config::GroupDependencyError: Gitolite::Config::GroupDependencyError
from /Users/matt/.rbenv/versions/1.9.2-p290/lib/ruby/gems/1.9.1/gems/gitolite-1.0.0/lib/gitolite/config.rb:223:in `build_groups_depgraph'
from /Users/matt/.rbenv/versions/1.9.2-p290/lib/ruby/gems/1.9.1/gems/gitolite-1.0.0/lib/gitolite/config.rb:71:in `block in to_file'
from /Users/matt/.rbenv/versions/1.9.2-p290/lib/ruby/gems/1.9.1/gems/gitolite-1.0.0/lib/gitolite/config.rb:69:in `open'
from /Users/matt/.rbenv/versions/1.9.2-p290/lib/ruby/gems/1.9.1/gems/gitolite-1.0.0/lib/gitolite/config.rb:69:in `to_file'
from /Users/matt/.rbenv/versions/1.9.2-p290/lib/ruby/gems/1.9.1/gems/gitolite-1.0.0/lib/gitolite/gitolite_admin.rb:69:in `block in save'
from /Users/matt/.rbenv/versions/1.9.2-p290/lib/ruby/gems/1.9.1/gems/gitolite-1.0.0/lib/gitolite/gitolite_admin.rb:67:in `chdir'
from /Users/matt/.rbenv/versions/1.9.2-p290/lib/ruby/gems/1.9.1/gems/gitolite-1.0.0/lib/gitolite/gitolite_admin.rb:67:in `save'
from (irb):9
from /Users/matt/.rbenv/versions/1.9.2-p290/bin/irb:12:in `<main>'
```

**Comment by wingrunr21** [comment]
Can I see the gitolite config?  Specifically where the groups are defined.  That error is thrown when the gem can't figure out a way to resolve group dependencies.

**Comment by mmcc** [comment]
Sorry for the delay, Graduation crap kept me busy all weekend. I'm going to post the groups and then an example of one repo, hopefully that will be enough.

```
@devs               = Andrew Charlie Raj Stephanie brandon codebot matt starscream
@admins             = Andrew Charlie Raj Stephanie codebot matt
@standard           = advisor boswell totaltech11
@internal_projects  = plx plexbash svn2git rails3_template_new
repo    atlas
  RW+                            = @devs
```

**Comment by wingrunr21** [comment]
Ok, I will take a look at this later today.

**Comment by wingrunr21** [comment]
Resolved in 7e3b8436291fac3561abd64507217dddc92025f4

Gem release 1.0.1 contains the fix and is now on rubygems.  Thanks for the bug report!

**Comment by mmcc** [comment]
Glad I could help!
